spec/lib/coach/request_serializer_spec.rb

```diff
@@ -34,5 +36,38 @@
         expect(sanitized).to eq('value')
       end
     end
+
+    context "as an instance" do
+      let(:mock_request) do
+        instance_double("Request", :request, format:              nil,
+                                             remote_ip:           nil,
+                                             uuid:                nil,
+                                             method:              nil,
+                                             filtered_parameters: nil,
+                                             filtered_env:        {
+                                               "foo" => "bar",
+                                               "HTTP_foo" => "bar"
+                                             })
+      end
+
+      subject(:request_serializer) { described_class.new(mock_request) }
+
+      describe "#serialize" do
+        subject(:serialized) { request_serializer.serialize }
+
+        it "rescues any exceptions request#fullpath may raise" do
+          allow(mock_request).to receive(:fullpath).and_raise
+
+          expect(serialized[:path]).to eq('unknown')
+        end
+
+        it "filters headers allowing only those prefixed with 'HTTP_'" do
```

Comment:
This makes a lot more sense I think, and is probably what you were going for last time.